xbmc/platform/darwin/tvos/TopShelf/ServiceProvider.mm

```diff
@@ -102,20 +101,27 @@ - (TVTopShelfContentStyle)topShelfStyle
     return contentItems;
   };
 
-  if ([movieArray count] > 0)
-  {
-    __auto_type itemMovie = [[TVContentItem alloc] initWithContentIdentifier:wrapperIdentifier];
-    itemMovie.title = [(sharedDict ?: shared) valueForKey:@"moviesTitle"];
-    itemMovie.topShelfItems = contentItemsFrom(movieArray);
-    [topShelfItems addObject:itemMovie];
-  }
+  // Add each category to TopShelf
+  __auto_type topShelfItems = [[NSMutableArray alloc] init];
 
-  if ([tvArray count] > 0)
+  for (id categoryKey in sharedDict)
```

Comment:
why `id`? Can it be not a `NSString`?

also use `enumerate...` method on `sharedDict` that gives you both keys and values.